src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -305,18 +360,62 @@ bool IsCreateHelper(IMethodSymbol method)
                     method.Parameters[0].Type.SpecialType == SpecialType.System_String;
         }
 
-        private static void AnalyzeTitle(IMethodSymbol descriptorCreationMethod, IFieldInitializerOperation creation, Action<Diagnostic> reportDiagnostic)
+        private static void AnalyzeTitle(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments, IFieldInitializerOperation creation)
+        {
+            IArgumentOperation titleArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("title", StringComparison.OrdinalIgnoreCase));
+
+            if (titleArgument != null &&
+                titleArgument.Parameter.Type != null &&
+                titleArgument.Parameter.Type.SpecialType == SpecialType.System_String)
+            {
+                operationAnalysisContext.ReportDiagnostic(creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString));
+
+                if (titleArgument.Value.ConstantValue.HasValue &&
+                    titleArgument.Value.ConstantValue.Value is string title &&
+                    (title.Contains(".") || ContainsLineReturn(title)))
+                {
+                    operationAnalysisContext.ReportDiagnostic(titleArgument.CreateDiagnostic(DefineDiagnosticTitleCorrectlyRule));
+                }
+            }
+        }
+
+        private static void AnalyzeMessage(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments)
+        {
+            IArgumentOperation messageArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("messageFormat", StringComparison.OrdinalIgnoreCase));
+
+            if (messageArgument != null &&
+                messageArgument.Value.ConstantValue.HasValue &&
+                messageArgument.Value.ConstantValue.Value is string message)
+            {
+                var splitPeriod = message.Split('.');
+                // Report if the message contains a line return or ends with '.' and is not a multi-sentences message
+                if ((splitPeriod.Length == 2 && splitPeriod[splitPeriod.Length - 1].Length == 0) ||
```

Comment:
Do we want to also ensure description ends with period when it is a multi-sentence?

Reply:
This check would need to be modified if prior Split argument is changed to ". ".

Reply:
> Do we want to also ensure message ends with period when it is a multi-sentence?

Seems reasonable.